pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/PersistentTopicsTest.java

```diff
@@ -389,6 +392,27 @@ public void testNonPartitionedTopics() {
                 0);
     }
 
+    @Test
+    public void testGetPartitionedMetadata() {
+
+        String testLocalTopicName = "test-topic";
+
+        // 1) Auto create non-partitioned topic.
+        PartitionedTopicMetadata partitionedMetadata = persistentTopics.getPartitionedMetadata(testTenant, testNamespace, testLocalTopicName, true, true);
+        Assert.assertEquals(0, partitionedMetadata.partitions);
+
+        // 2. Add validate policy assert.
+        doThrow(new RestException(Response.Status.FORBIDDEN, "mock message"))
+                .when(persistentTopics).validateTopicPolicyOperation(persistentTopics.topicName, PolicyName.PARTITION, PolicyOperation.WRITE);
+        try {
+            persistentTopics.getPartitionedMetadata(testTenant, testNamespace, testLocalTopicName, true, true);
+            Assert.fail("Should fail validation on policy exception");
+        } catch (RestException e) {
+            Assert.assertEquals(Response.Status.FORBIDDEN.getStatusCode(), e.getResponse().getStatus());
+        }
+
```

Comment:
Could we add tests to when the param `checkAllowAutoCreation` is `false`?

Reply:
Good idea, I added.